hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinySessionFactoryImpl.java

```diff
@@ -57,6 +58,32 @@ public MutinySessionFactoryImpl(SessionFactoryImpl delegate) {
 				.map( MutinySessionImpl::new );
 	}
 
+	@Override
+	public Mutiny.StatelessSession createStatelessSession() {
```

Comment:
I see this isn't newly introduced on the `StatelessSession` as you have it on `Session` too, but I don't fully understand why we need two different APIs to open/create a XSession.

Shouldn't all connection acquisition be lazy, or left as a configurable policy as in Hibernate ORM? The current policy contract evolved from integration needs with various containers and datasource pools. In Quarkus specifically, we're working to have it eventually be able to use a strategy of "acquire deferred + release just-before-commit" which is the most efficient strategy.

I'm not suggesting we implement the same policy now in Reactive, but I'd prefer the API we choose to expose will leave us some margin of flexibility to improve on this aspect as necessary.

I'll open a separate issue to discuss this.

Reply:
link to #316

Reply:
I've responded in the issue.